src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -629,8 +629,31 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             return ret;
         }
 
-        private static Encoding GetEncoding(int codePage)
+        private static Encoding GetConsoleEncodingOrThrow(ConsoleStreamDirection encodingType)
         {
+            const int INVALID_HANDLE = 6;
+            int codePage;
+            try
+            {
+                codePage = (encodingType == ConsoleStreamDirection.Input) ?
+                    (int)Interop.Kernel32.GetConsoleCP() :
+                    (int)Interop.Kernel32.GetConsoleOutputCP();
+
+                int error = Marshal.GetLastWin32Error();
+                if (error == INVALID_HANDLE)
```

Comment:
Thanks @JeremyKuhne for confirming - I assumed it does since I haven't seen failing API which doesn't before - we could also potentially check for return value 0